pkg/virt-controller/watch/snapshot.go

```diff
@@ -637,6 +637,46 @@ func (ctrl *SnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Virt
 	return nil
 }
 
+func (ctrl *SnapshotController) updateVolumeSnapshotStatuses(vm *kubevirtv1.VirtualMachine) error {
+	log.Log.V(3).Infof("Update volume snapshot status for VM [%s/%s]", vm.Namespace, vm.Name)
+
+	var statuses []kubevirtv1.VolumeSnapshotStatus
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		log.Log.V(3).Infof("Update volume snapshot status for volume [%s]", volume.Name)
+		status := ctrl.getVolumeSnapshotStatus(vm, &volume)
+		statuses = append(statuses, status)
+	}
+
+	vm.Status.VolumeSnapshotStatuses = statuses
```

Comment:
Updating the VM in memory is not sufficient to persist the changes.  You have to call the `VMStatusUpdater`

Reply:
also do a deep copy before mutating any resource you get from an informer

Reply:
Hmm, I can see the changes with `kubevirt describe vm` (not immediately though).

Reply:
the informer cache is shared so some other controller is probably updating the VM and persisting your changes along with it